integration/tests/cook/test_client.py

```diff
@@ -56,3 +59,37 @@ def test_kill(self):
         self.client.kill(uuid)
         job = self.client.query(uuid)
         self.assertEqual(job.status, JobStatus.COMPLETED)
+
+    def test_container_submit(self):
+        container = DockerContainer(util.docker_image())
+        self.assertIsNotNone(container.image)
+        uuid = self.client.submit(command='ls',
+                                  container=DockerContainer(image))
+        job = self.client.query(uuid)
+
+        remote_container = job.container
+        self.assertEqual(remote_container['type'], 'DOCKER')
+        self.assertEqual(remote_container['docker']['image'], container.image)
+
+        self.client.kill(uuid)
+
+    def test_instance_query(self):
+        """Test that parsing an instance yielded from Cook works."""
+        uuid = self.client.submit(command=f'sleep {util.DEFAULT_TEST_TIMEOUT_SECS}',
+                                  cpus=0.5,
+                                  mem=1.0,
+                                  max_retries=5)
+
+        def is_instance_available(job: Job):
+            any_instances = len(job.instances) > 0
+            if not any_instances:
+                util.logger.info(f"Job {job.uuid} has no instances yet.")
+            return any_instances
+            
+        util.wait_until(lambda: self.client.query(uuid),
+                        is_instance_available)
```

Comment:
This can simplify to just:

```suggestion
        util.wait_for_instance(type(self).cook_url, uuid)
```